gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/internal/WarnAboutComposeWithSerialization.kt

```diff
@@ -0,0 +1,28 @@
+package org.jetbrains.compose.internal
+
+import org.gradle.api.Project
+import org.gradle.configurationcache.extensions.serviceOf
+import org.gradle.internal.logging.text.StyledTextOutput
+import org.gradle.internal.logging.text.StyledTextOutputFactory
+
+internal fun Project.checkAndWarnAboutComposeWithSerialization() {
+    afterEvaluate {
+        val usesKotlinxSerialization = configurations.names.asSequence().filter {
+            it.startsWith("kotlinCompilerPluginClasspath")
+        }.any { configurationName ->
+            configurations.getByName(configurationName).dependencies.any { dependency ->
+                dependency.name.contains("kotlin-serialization")
+            }
+        }
+
+        if (usesKotlinxSerialization) {
+            val out = serviceOf<StyledTextOutputFactory>().create("COMPOSE_PLUGIN")
```

Comment:
Please don't use internal apis for simple logging, `project.logger.warn()` or `project.logger.error()` are more than enough